homeassistant/components/cover/mqtt.py

```diff
@@ -303,7 +341,8 @@ def supported_features(self):
         if self._command_topic is not None:
             supported_features = OPEN_CLOSE_FEATURES
 
-        if self._position_topic is not None:
+        if (self._set_position_topic is not None and
+                self._get_position_topic is not None):
```

Comment:
If we require both topics to be set to support set position, shouldn't we require both topics in the config validation schema? Is a case where only either is set supported?

Reply:
`get_position_topic` can work without `set_position_topic`.
I've added checking of get_position_topic for SUPPORT_SET_POSITION because if user would set
```
optimistic: true
state_topic: 'state-topic'
set_position_topic: 'position-topic'
```
for me it wouldn't be logical that `set_cover_position` would be the only function that changes self._position and this slider would appear for him somehow... 
This way we will break this new functionality that if only state-topic is set then position is not available.

Reply:
If we don't support setting set_position_topic without get_position_topic we should add a custom validator that checks this in the config schema.

Reply:
Validator looks good! So now we don't need to check both topics here right?